tools/run_tests/run_tests.py

```diff
@@ -78,6 +77,14 @@ def platform_string():
 _DEFAULT_TIMEOUT_SECONDS = 5 * 60
 
 
+# TODO (https://github./grpc/grpc/issues/7934): An object which when passed
+# as a compiler option to the script along with the travis flag, ensures that
+# the tests pass on any language. This approach will no longer be necessary
+# when the testing infrastructure would be able to support different testing
+# configurations for different branches of the repository.
+_FREE_PASS = object()
```

Comment:
the name "FREE_PASS" could be more descriptive. 
To illustrate this, look at the `if self.args.compiler is _FREE_PASS and self.args.travis:` line pretending you don't know the context and you will be really wondering what the heck is "FREE_PASS".

How about
_UNSUPPORTED_COMPILER_SKIP_TESTS?